onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -0,0 +1,571 @@
+        return false;
+      }
+
+      //String data type is not supported
+      const auto* type_proto = node->InputDefs()[0]->TypeAsProto();
+      if (type_proto->tensor_type().elem_type() == ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_STRING) {
+        return false;
+      }
+
+      auto graph_inputs = graph_viewer.GetInputs();
+      auto it = find(graph_inputs.begin(), graph_inputs.end(), node->InputDefs()[0]);
+      if (it != graph_inputs.end()) {
+        if (input_dims == 3 || input_dims == 2 || input_dims > 5) {
+          return false;
+        }
+      }
+    }
+
+    if (node->OpType() == "Unsqueeze") {
+      auto graph_inputs = graph_viewer.GetInputs();
+      auto attributes = node->GetAttributes();
+      auto axes = attributes["axes"].ints();
+      auto it = find(graph_inputs.begin(), graph_inputs.end(), node->InputDefs()[0]);
+      if (it != graph_inputs.end()) {
+        if (input_dims + axes.size() > 5) {
+          return false;
+        }
+      }
+      const auto* type_proto = node->InputDefs()[0]->TypeAsProto();
+      if (type_proto->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT)
+        return false;
+    }
+
+    //Don't support Reshape without output dims
+    if (node->OpType() == "Reshape") {
+      if (output_dims == 0) {
+        return false;
+      }
+    }
+
+    //Broadcasting is not supported for Mul, Add and Sum
+    if (node->OpType() == "Mul" || node->OpType() == "Add" || node->OpType() == "Sum") {
+      auto graph_inputs = graph_viewer.GetInputs();
+      auto it = find(graph_inputs.begin(), graph_inputs.end(), node->InputDefs()[0]);
+
+      if (it != graph_inputs.end()) {
+        size_t dims = input_arrays[0].size();
+        if (dims == 0)
+          return false;
+
+        for (int i = 0; i < num_inputs; i++) {
+          if (dims != input_arrays[i].size())
+            return false;
+        }
+        for (int i = 0; i < num_inputs; i++) {
+          for (size_t j = 0; j < dims; j++) {
+            if (input_arrays[0][j] != input_arrays[i][j]) {
+            //   isGraphSupported = false;
+                return false;
+            //   break;
+            }
+          }
+        }
+      }
+    }
+
+    if (node->OpType() == "Softmax") {
+      auto graph_inputs = graph_viewer.GetInputs();
+      auto it = find(graph_inputs.begin(), graph_inputs.end(), node->InputDefs()[0]);
+      if (it != graph_inputs.end()) {
+        if (input_dims != 2) {
+          return false;
+        }
+      }
+      auto attributes = node->GetAttributes();
+      auto axis = attributes["axis"].i();
+      if (axis != 1)
+        return false;
+    }
+
+    if (node->OpType() == "Flatten") {
+      auto attributes = node->GetAttributes();
+      auto axis = attributes["axis"].i();
+      if (dev_id == "MYRIAD" || dev_id == "HDDL") {
+        if (axis != 1)
+          return false;
+      }
+    }
+  }
+  //Disable Resnet DUC for MYRIAD and HDDL
```

Comment:
would be better if we don't mention specific models by name?
gives the impression this is not a generalized solution.

Reply:
Removed the comment